crates/astria-sequencer/src/app.rs

```diff
@@ -442,133 +507,103 @@ impl App {
             .get_chain_id()
             .await
-            .context("failed to get chain ID from state")?
-            .try_into()
-            .context("invalid chain ID")?;
+            .context("failed to get chain ID from state")?;
 
         // call begin_block on all components
         // NOTE: the fields marked `unused` are not used by any of the components;
         // however, we need to still construct a `BeginBlock` type for now as
         // the penumbra IBC implementation still requires it as a parameter.
-        let begin_block = abci::request::BeginBlock {
-            hash: finalize_block.hash,
-            byzantine_validators: finalize_block.misbehavior.clone(),
+        let begin_block: abci::request::BeginBlock = abci::request::BeginBlock {
+            hash: Hash::default(), // unused
+            byzantine_validators: block_data.misbehavior.clone(),
             header: Header {
                 app_hash: self.app_hash.clone(),
                 chain_id: chain_id.clone(),
-                consensus_hash: Hash::default(), // unused
-                data_hash: Some(Hash::try_from(data_hash.to_vec()).unwrap()),
+                consensus_hash: Hash::default(),      // unused
+                data_hash: Some(Hash::default()),     // unused
                 evidence_hash: Some(Hash::default()), // unused
-                height: finalize_block.height,
+                height: block_data.height,
                 last_block_id: None,                      // unused
                 last_commit_hash: Some(Hash::default()),  // unused
                 last_results_hash: Some(Hash::default()), // unused
-                next_validators_hash: finalize_block.next_validators_hash,
-                proposer_address: finalize_block.proposer_address,
-                time: finalize_block.time,
+                next_validators_hash: block_data.next_validators_hash,
+                proposer_address: block_data.proposer_address,
+                time: block_data.time,
                 validators_hash: Hash::default(), // unused
                 version: tendermint::block::header::Version {
                     // unused
                     app: 0,
                     block: 0,
                 },
             },
-            last_commit_info: finalize_block.decided_last_commit.clone(),
+            last_commit_info: tendermint::abci::types::CommitInfo {
+                round: 0u16.into(), // unused
+                votes: vec![],
+            }, // unused
         };
 
-        self.begin_block(&begin_block, storage.clone())
+        self.begin_block(&begin_block)
             .await
             .context("failed to call begin_block")?;
 
+        Ok(())
+    }
+
+    /// Executes the given block, but does not write it to disk.
+    ///
+    /// `commit` must be called after this to write the block to disk.
+    ///
+    /// This is called by cometbft after the block has already been
+    /// finalized by the network.
```

Comment:
nit: duplication of finalize here kinda confusing, more technically correct to say "consensus reached" perhaps?